derivation.nix

```diff
@@ -37,17 +45,40 @@ in
     , ...
     }@attrs:
     let
+      isWorkspace = workspace != null && components != [];
       nativeBuildInputs = [
         nodejs
         pnpm
         pkg-config
       ] ++ extraBuildInputs;
+      computedNodeModuleSources =
+        (if pnpmWorkspaceYaml == null
+          then []
+          else [
+            {name = "pnpm-workspace.yaml"; value = pnpmWorkspaceYaml;}
+          ]
+        ) ++ extraNodeModuleSources;
+      copyLink =
+        if copyNodeModules
+          then "cp -r"
+          else "ln -s";
```

Comment:
All of the definitions here are just helpers to simplify the changes below. So is `isWorkspace`.